### ISSUE TITLE
Allow scripted dashboard directory to be configurable

### ISSUE THREAD

**Issue by CVTJNII** [grafana/grafana#3809]
Currently scripted dashboards go in public/dashboards/.  This directory isn't unique to scripted dashboards, JSON config files for the home dashboard as well as several other files also need to reside in this directory.  This is causing problems in our environment as we are treating the JSON files as part of the underlying install, but need to control user added dashboard scripts as well, and having both in the same directory is troublesome.  We're running Grafana in Docker and having both in the same directory means I cannot simply mount a volume of scripted dashboards into the container, I need to have some way to merge the JSON files with the scripts to have a single directory Grafana can read, and this is difficult and painful.

It would be very nice for the scripted dashboards directory to be configurable.  This way legacy behavior of the shared directory can be retained, but those like me who need separate directories can tune the dashboard directory off to another point in the filesystem.

Also, until the core request above is resolved, I would greatly appreciate some pointers into where scripted dashboards are handled in the code.  Patching Grafana to use a different directory for scripted dashboards is the path of least resistance for getting Grafana up in our environment, but I for the life of me can't find what maps the /dashboard/script/ URI to the public/dashboards/ directory on the filesystem.  A pointers would be appreciated.  Thanks.

**Comment by CVTJNII** [comment]
I hit on a simple workaround for this.  As we're already proxying Grafana behind Nginx I simply added a 'location ~ ^/public/dashboards/(.*\.js)$' directive to my Nginx config and aliased the path off to somewhere else in Nginx.  As the directory in question ultimately served by Go with the actual processing happening on the client in Java script, this allows me to transparently carve the scripted dashboards off to another location without fiddling with or recompiling Grafana.

**Comment by arcolife** [comment]
+1

**Comment by ajorgensen** [comment]
I have a need for this as well and have started putting together a solution. Just wanted to let people incase someone else is working on this in parallel or has already done substantial work on it.

**Comment by dlockman** [comment]
> we are deprecating scripted dashboards (some time this year), will be replaced with a new type of dashboard and use the new provisioning system. So this issue will never be implemented.

Is there a longer description of the reasoning behind this, and what will replace it ("the new provisioning system") somewhere publicly available? I've been looking around in lots of Issue threads with no success, and the official Docs and README still list scripted dashboards as fully supported.

We're heading down the road of scripted dashboards at the moment to solve some problems, and would like to stay as aligned with future Grafana plans as possible.

**Comment by iroller** [comment]
Also interested if there're any discussions available to public on a new type of dashboards and the future of scripted dashboards.

**Comment by kenjura** [comment]
It's been over a year, and I'm interested in knowing what the replacement to scripted dashboards will be. I'm implementing a solution using provisioning and containers, and it's working swimmingly...except for the scripted dashboard.
